pkg/virt-controller/watch/migration.go

```diff
@@ -609,6 +609,24 @@ func (c *MigrationController) createTargetPod(migration *virtv1.VirtualMachineIn
 		}
 	}
 
+	// The target pod may shares ressources with the sources pod (RWX disks for example)
+	// Therefore, it needs to share the same SELinux categories to inherit the same permissions
+	// Note: there is a small probablility that the target pod will share the same categories as another pod on its node.
+	//   It is a slight security concern, but not as bad as removing categories on all shared objects for the duration of the migration.
+	if vmi.Status.SelinuxContext == "" {
+		return fmt.Errorf("SELinux context not present on VMI")
+	}
+	seFields := strings.SplitN(vmi.Status.SelinuxContext, ":", 4)
+	if len(seFields) == 4 {
+		// The SELinux context looks like "system_u:object_r:container_file_t:s0:c1,c2", we care about "s0:c1,c2"
+		if templatePod.Spec.SecurityContext == nil {
+			templatePod.Spec.SecurityContext = &k8sv1.PodSecurityContext{}
+		}
+		templatePod.Spec.SecurityContext.SELinuxOptions = &k8sv1.SELinuxOptions{
+			Level: seFields[3],
```

Comment:
just a suggestion, may be worth to try construct the context and grab the level out of it?
https://github.com/kubevirt/kubevirt/blob/897d63c6d565328e07991a36b1e81292260db4a4/vendor/github.com/opencontainers/selinux/go-selinux/selinux_linux.go#L715-L721

Reply:
Good point, will do!